.gitlab/docker/Dockerfile.spack_base

```diff
@@ -33,9 +33,11 @@ RUN apt-get update && apt-get -yqq install --no-install-recommends \
 # Install spack
 ARG SPACK_REPO=https://github.com/spack/spack
 ARG SPACK_VERSION
+ARG SPACK_COMMIT
 ENV SPACK_ROOT=/opt/spack
 RUN mkdir -p $SPACK_ROOT \
-    && git clone --depth 1 -b ${SPACK_VERSION} ${SPACK_REPO} ${SPACK_ROOT}
+    && git clone --depth 1 -b ${SPACK_VERSION} ${SPACK_REPO} ${SPACK_ROOT} \
+    && git reset --hard ${SPACK_COMMIT}
```

Comment:
Perhaps we use only `SPACK_COMMIT` and pull the tarball instead for `SPACK_COMMIT`? Should be both faster and `SPACK_VERSION` doesn't really say anything if we're specifying the commit explicitly (plus they can become out of sync).